x-pack/test/apm_api_integration/tests/diagnostics/apm_events.spec.ts

```diff
@@ -97,6 +98,60 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         ]);
       });
 
+      describe('transactions', async () => {
+        let body: APIReturnType<'GET /internal/apm/diagnostics'>;
+
+        const expectedDocCount = 450;
+
+        beforeEach(async () => {
+          const res = await apmApiClient.adminUser({
+            endpoint: 'GET /internal/apm/diagnostics',
+            params: {
+              query: { start: new Date(start).toISOString(), end: new Date(end).toISOString() },
+            },
+          });
+
+          body = res.body;
+        });
+
+        it('raw transaction events', () => {
+          const rawTransactions = body.apmEvents.find(({ kuery }) =>
+            kuery.includes('processor.event: "transaction"')
+          );
+
+          expect(rawTransactions?.docCount).to.be(expectedDocCount);
+        });
+
+        it('transaction metrics', () => {
+          const transactionMetrics = body.apmEvents.find(({ kuery }) =>
+            kuery.includes('metricset.name: "transaction"')
+          );
+
+          expect(transactionMetrics?.docCount).to.be(21);
```

Comment:
Do you think checking the sum of the internals to match the total doc count will  add value? 

For example, 15 + 4 + 2 = 21